### PR TITLE
Code Conventions

### PR DESCRIPTION
**Pull request by brentonannan**

### CODE REVIEW

**Comment by martinstannard** [comment]
nice!

**Comment by kunzmann** [comment]
:+1:

File: code-conventions/README.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+      def call
+        if has_capacity?
+          new_class = Service::CreateSchoolClass.new(teacher, product).call
+          original_class.students.each do |student|
+            # Adds student to class, notifies them
+            Workflow::SchoolClass::AddStudent.new(new_class, student).call
+          end
+          Service::SchoolClass::Created::NotifySubCo.new(new_class).call
+          return new_class
```
Comment (brentonannan) [inline code-conventions/README.md]: I like me a good explicit `return`.

Comment (martinstannard) [inline code-conventions/README.md]: no

Comment (kunzmann) [inline code-conventions/README.md]: :boom::gun::point_down:

Comment (brentonannan) [inline code-conventions/README.md]: Reasoning?

Comment (stuart) [inline code-conventions/README.md]: Ruby always returns the last thing. It's a nice convention. 
To me an explicit return says, "hey we are returning early and breaking out of the normal flow of this
method for some reason."
Plus it takes up 7 extra bytes.

Comment (martinstannard) [inline code-conventions/README.md]: And when do you do it and when don't you (unless for the breakout case)?

Comment (brentonannan) [inline code-conventions/README.md]: I know that it does that, and I know the convention, and use it the majority of the time.

If, however, I've set the thing I want to return as a variable early on, I feel like just writing the name of the variable at the end of what I'm doing doesn't actually convey any intention whatsoever. It's like some guy just standing in a field, yelling stuff out of context - you don't know that he's trying to give you a message, unless you stop and think "Why is he just blatting that variable there? Oh, it's what he wants to return."

Explicitly returning is saying "I made this thing before - you can see where I made it, you can see all the wonderful things I've done with it, now I want you to notice that if you use this method, you're going to get that thing back, not something else." IMO.

**Comment by martinstannard** [comment]
I'm happy with this as a start, with a general note about failure handling. We can expand/cleanup in subsequent PRs. :shipit:

**Comment by brentonannan** [comment]
Still got a bunch of different types to write about, though perhaps I'll do those as separate PRs.

File: code-conventions/README.md (hunk 1 above)
Comment (JamesFerguson) [inline code-conventions/README.md]: I agree with your reasoning and your aesthetic @brentonannan, but I think this convention was established so early and so strictly that it's entrenched to the point where trying to change it, even with good reason, would be much more work than just going with the flow.

**Pull request merged**